📊 data: undp 2024

lucasrodes opened the pull request:
[↑ Tracking issue](https://github.com/owid/owid-issues/issues/1291)

- Add the new UNDP 2024 dataset to ETL.
- Update all charts (32) in the staging server.

lucasrodes commented:
⚠️ WIP

## Large changes

### Expected years of schooling
- Lebanon
- Marshall Islands
- Moldova


### Gender Inequality Index
- Cyprus

New Data: Somalia
Removed data: Central African Republic, South Sudan, Turkmenistan

### Gross national income per capita
- Argentina
- Bangladesh
- Afghanistan
- Libya

lucasrodes commented:
⚠️ WIP

- Why is it only showing data for 2022? http://staging-site-data-undp/admin/charts/7135/edit

paarriagadap commented on etl/steps/data/garden/un/2024-04-09/undp_hdr.meta.yml:
```diff
@@ -0,0 +1,463 @@
+definitions:
```
For the previous version (`2022-11-29`) I added additional metadata in the new format in the grapher step. It would be good to integrate that to this file, since it was agreed in the HDI redesign.

The differences are mostly around the topic tags and the metadata and grapher config for HDI.

paarriagadap reviewed (changes requested):
Looks good, thank you! I request changes to integrate the metadata we included in the HDI article we worked last year with Bastian. You can find that in `grapher/un/2022-11-29/undp_hdr.meta.yml`

How different are the estimates you listed? If it's not too much it's not a big deal.

paarriagadap commented on etl/steps/data/garden/un/2024-04-09/undp_hdr.meta.yml:
```diff
@@ -0,0 +1,463 @@
+definitions:
+  common:
```
`processing_level` is missing

paarriagadap commented on apps/wizard/pages/charts/variable_config.py:
```diff
@@ -311,12 +312,13 @@ def ask_and_get_variable_mapping(search_form) -> "VariableConfig":
-                    "Something went wrong! This can be due to several reasons: One (or both) of the variables are not numeric, `values` for one of the variables does not have the columns `entityName` and `year`. Please check the error message below. Report the error #002001"
-                )
+            plot_comparison_two_variables(df_data, variable_old, variable_new, variable_id_to_display)  # type: ignore
+            # try:
```
Delete this comment if no longer needed

lucasrodes commented:
@paarriagadap thanks for all your suggestions! I addressed all your comments. They all make sense to me.

Could you double-check the metadata? I've integrated the metadata from the old Grapher step. I've also added some more details to female/male dimensions of the variable (and others).

lucasrodes commented:
I'll merge for now, we can update and modify the metadata later on if needed.
